baybe/acquisition/base.py

```diff
@@ -24,15 +28,37 @@ class AcquisitionFunction(ABC, SerialMixin):
     _abbreviation: ClassVar[str]
     """An alternative name for type resolution."""
 
-    def to_botorch(self, surrogate: Surrogate, best_f: float):
+    @classproperty
+    def is_mc(cls) -> bool:
+        """Flag indicating whether this is a Monte-Carlo acquisition function."""
+        return cls._abbreviation.startswith("q")
+
+    def to_botorch(
+        self,
+        surrogate: Surrogate,
+        train_x: pd.DataFrame,
+        train_y: pd.DataFrame,
+    ):
         """Create the botorch-ready representation of the function."""
-        import botorch.acquisition as botorch_acquisition
+        import botorch.acquisition as botorch_analytical_acqf
 
-        from baybe.acquisition.adapter import debotorchize
+        from baybe.acquisition.adapter import _AdapterModel
 
-        acqf_cls = getattr(botorch_acquisition, self.__class__.__name__)
+        acqf_cls = getattr(botorch_analytical_acqf, self.__class__.__name__)
+        params_dict = filter_attributes(object=self, callable_=acqf_cls.__init__)
 
-        return debotorchize(acqf_cls)(surrogate, best_f)
+        additional_params = {
+            p: v
+            for p, v in {
+                "model": _AdapterModel(surrogate),
+                "best_f": train_y.max().item(),
+                "X_baseline": to_tensor(train_x),
```

Comment:
Can you quickly comment on this `X_baseline` thing? Have no clue what it actually does nor why it needs to be additionally passed...

Reply:
its a required arg for some of the qNoisy variants
Its supposed to the the points that are likely the best points
COntrary to my initial thinking however you can simply pass here all candidates, botorch prunes them to only contain the likely best points if `prune_baselins` option (part of the acqf attributes) is True

Reply:
Can you point me more specifically to where you (think you) saw this? Asking because:
* For the two acqfs that we have that use it, I can see that `X_baseline` is optional
* Interestingly, the default auf `prune_baseline` is different in both cases.

That said, if we don't have to, I'd say we rather don't bother with it for now (both `prune_baseline` and `X_baseline` and just resort defaults), removing both the argument and the field. 

But let's hear your opinion first.

Reply:
Am surprised by your statement that its optional, according to the constructors its not. Also if I uncomment the `X_Baseline` line, `test_iterations` with exactly these two acqfs fails saying they're missing `X_baseline`, so are you sure?

Furthermore, I am just doing what the factory does, hence that should be gucci. Also a default `prune_baseline=True` makes sense to me

Reply:
Sorry, only looked at `prune_baseline`, so just forget what I said. Also makes sense that `X_baseline` is mandatory, now that I read what it actually is 😄 So let me close this thread